lib/wit/package.py

```diff
@@ -122,14 +127,27 @@ def add_dependency(self):
 
     def checkout(self, wsroot):
         """Move to root directory and checkout"""
-        shutil.move(str(self.repo.get_path()), str(wsroot/self.name))
+        current_origin = self.repo.get_remote()
+        wanted_origin = self.source
+        if current_origin != wanted_origin:
+            if self.repo.path.parts[-2] == '.wit':
+                self.repo.set_origin(self.source)
+            else:
+                log.warn("Package '{}' wants a different git remote origin.\n"
+                         "Origin is currently:\n  {}\n"
+                         "Package '{}' wants origin:\n  {}\n"
+                         "Please manually update the origin with:\n"
+                         "  git -C {} \\\n    remote set-url origin {}"
+                         "".format(self.name, current_origin, self.name,
+                                   wanted_origin, self.repo.path, wanted_origin))
```

Comment:
So this warning will occur if you have a workspace, and then change it in such a way that we introduce a dependency on a newer version of a repo with a different remote?

Reply:
I think a test of this behavior would be good

Reply:
> So this warning will occur if you have a workspace, and then change it in such a way that we introduce a dependency on a newer version of a repo with a different remote?

Exactly.

> I think a test of this behavior would be good

 Will do.

Reply:
> I think a test of this behavior would be good

Fixed by https://github.com/sifive/wit/pull/135/commits/cab246f18a394d8cbcc47e0fdeec44793781d074